src/restconfig/src/main/java/org/geoserver/catalog/rest/StyleFinder.java

```diff
@@ -99,8 +99,12 @@ public void onApplicationEvent(ApplicationEvent event) {
         if (event instanceof ContextLoadedEvent) {
             // register style format mime types
             for (StyleHandler sh : Styles.handlers()) {
-                Version ver = sh.getVersions().iterator().next();
-                MediaTypes.registerExtension(sh.getFileExtension(), new MediaType(sh.mimeType(ver)));
+                // Only map the StyleHandler's extension to its mime type 
+                // if the extension does not conflict with existing mappings.
+                if (MediaTypes.getMediaTypeForExtension(sh.getFileExtension()) == null) {
```

Comment:
Very minor, but a class-level javadoc (or other sort of documentation that is more visible than an inline comment) mentioning this behavior might be a good idea. I could imagine in the future someone trying to add a new style format and it "inexplicibly" not getting picked up.